development/extensions/tutorial_modules.rst

```diff
@@ -0,0 +1,319 @@
+=================
+Tutorial: Modules
+=================
+
+Introduction
+============
+
+In phpBB you can create your own modules for the User, Moderator and
+Administration Control Panels. This modular approach has many advantages
+over creating settings pages from scratch:
+
+  * You can manage modules in the ACP.
+  * You don't have to directly handle authorization.
+  * You get a navigation tree built-in to the system.
+  * The module templates already contain features like pagination,
+    breadcrumbs and confirmation messages.
+
+This tutorial explains:
+
+ * `Creating Control Panel Modules`_
+ * `Installing Control Panel Modules`_
+
+.. note::
+    This tutorial describes how to create an ACP module. If you want to
+    create MCP or UCP modules, you usually only need to replace all occurrences
+    of "acp" with "mcp" or "ucp".
+
+
+Creating Control Panel Modules
+==============================
+
+A module requires three files:
+
+ 1. `Module info`_
+ 2. `Module class`_
+ 3. `Module template`_
+
+Module info
+-----------
+
+The module info class simply contains a method that returns an array
+of information describing the module. Information like the name of
+the module, available modes and required permissions.
+
+This file must always end in ``*_info.php`` and
+should be stored in the ``acp/`` folder of the extension.
+For this tutorial, we will use ``ext/acme/demo/acp/main_info.php``:
+
+.. code-block:: php
+
+    <?php
+
+    namespace acme\demo\acp;
+
+    class main_info
+    {
+        public function module()
+        {
+            return array(
+                'filename'  => '\acme\demo\acp\main_module',
+                'title'     => 'ACP_DEMO_TITLE',
+                'modes'    => array(
+                    'settings'  => array(
+                        'title' => 'ACP_DEMO',
+                        'auth'  => 'ext_acme/demo && acl_a_board',
+                        'cat'   => array('ACP_DEMO_TITLE'),
+                    ),
+                ),
+            );
+        }
+    }
+
+The ``main_info`` class has only one method ``module()`` which returns an array
+with the following information:
+
+.. csv-table::
+    :header: "Key", "Content"
+    :delim: |
+
+    ``filename`` | "Fully name-spaced path to the `Module class`_, starting with a leading slash."
+    ``title`` | "Language key for the module-title that is displayed in the ACP module
+    management section."
+    ``modes`` | "Contains an array of modes for the module. See the next table
+    for more details."
+
+Note that we have defined one mode named ``settings`` with the following information:
+
+.. csv-table::
+    :header: "Key", "Content"
+    :delim: |
+
+    ``title`` | "Language key of the mode-title that is displayed in the ACP module
+    management section."
+    ``auth`` | "An authorisation key used to control access to the mode.
+    In this case we require the extension ``acme/demo`` to be enabled and
```

Comment:
say something regarding that the acme/demo auth is required, and is not a optional auth.